src/Serialization/Rings.jl

```diff
@@ -0,0 +1,125 @@
+################################################################################
+#  non simpleton base rings
+
+function load_internal(s::DeserializerState, t::Type{Nemo.NmodRing}, dict::Dict)
+    n = load_type_dispatch(s, UInt64, dict[:n])
+    
+    return t(n)
+end
+
+function save_internal(s::SerializerState, R::Type{Nemo.NmodRing})
+    return Dict(
+        :n => save_type_dispatch(s, R.n)
+    )
+end
+
+################################################################################
+# Multi  Variate Polynomial Ring
+function save_internal(s::SerializerState, R::MPolyRing)
+    base_ring = R.base_ring
+    return Dict(
+        :symbols => save_type_dispatch(s, R.S),
+        :base_ring => save_type_dispatch(s, base_ring),
+    )
+end
+
+function load_internal(s::DeserializerState, ::Type{<: MPolyRing}, dict::Dict)
+    base_ring = load_type_dispatch(s, dict[:base_ring], check_namespace=false)
+    symbols = load_type_dispatch(s, Vector{Symbol}, dict[:symbols]) 
+    
+    return PolynomialRing(base_ring, symbols)
+end
+
+################################################################################
+# Uni Variate Polynomial Ring
+function save_internal(s::SerializerState, R::PolyRing)
+    base_ring = R.base_ring
+    return Dict(
+        :symbol => save_type_dispatch(s, R.S),
+        :base_ring => save_type_dispatch(s, base_ring),
+    )
```

Comment:
Please don't access members of structs directly if you can avoid it; this hinders future refactoring.
```suggestion
    return Dict(
        :symbol => save_type_dispatch(s, symbols(R)),
        :base_ring => save_type_dispatch(s, base_ring(R)),
    )
```
BTW, any reason you call the key `symbol` instead of `symbols` ?